Content.IntegrationTests/Tests/CargoTests.cs

```diff
@@ -0,0 +1,37 @@
+using System.Threading.Tasks;
+using Content.Server.Cargo.Components;
+using Content.Shared.Cargo;
+using NUnit.Framework;
+using Robust.Shared.Prototypes;
+
+namespace Content.IntegrationTests.Tests;
+
+[TestFixture]
+public sealed class CargoTests : ContentIntegrationTest
+{
+    [Test]
+    public async Task TestGalacticMarketProducts()
+    {
+        var server = StartServerDummyTicker();
+        await server.WaitIdleAsync();
+
+        var protoManager = server.ResolveDependency<IPrototypeManager>();
+
+        await server.WaitAssertion(() =>
+        {
+            foreach (var ent in protoManager.EnumeratePrototypes<EntityPrototype>())
+            {
+                if (!ent.Components.TryGetValue("GalacticMarket", out var comp)) continue;
+
+                var market = (GalacticMarketComponent) comp;
+
+                foreach (var product in market.Products)
+                {
+                    // This top assert is probably caught by the TryIndex but juusssttt in case it ever gets reverted.
+                    Assert.That(protoManager.HasIndex<CargoProductPrototype>(product.ID), $"Unable to find cargo product for {product.ID} on prototype {ent}");
+                    Assert.That(protoManager.HasIndex<EntityPrototype>(product.Product), $"Unable to find cargo entity for {product.Product} on prototype {ent}");
```

Comment:
Given that this test is just checking whether the prototypes exist w/o doing anything more, couldn't you achieve the same thing by just adding
`customTypeSerializer: typeof(PrototypeIdListSerializer<CargoProductPrototype>))`
and
`customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))`
to the relevant datafields, so that the yaml linter checks them?

Reply:
That's the proper way of solving this, yup

Reply:
Woops, habit from EntityPrototype not being supported.

Reply:
Just gives me an esoteric error?